apollo-compiler/src/main/kotlin/com/apollostack/compiler/GraphqlCompiler.kt

```diff
@@ -1,6 +1,51 @@
 package com.apollostack.compiler
 
+import com.apollostack.compiler.ir.Field
+import com.apollostack.compiler.ir.QueryIntermediateRepresentation
+import com.squareup.javapoet.*
+import com.squareup.moshi.Moshi
+import java.io.File
+import javax.lang.model.element.Modifier
+
 open class GraphqlCompiler {
+  private val moshi = Moshi.Builder().build()
+
+  fun write(relativePath: String): Status {
+    val irAdapter = moshi.adapter(QueryIntermediateRepresentation::class.java)
+    val ir = irAdapter.fromJson(File(relativePath).readText())
+    // TODO: Handle multiple or no operations
+    val operation = ir.operations.first()
+    val methods = operation.fields.map { it.toMethodSpec() }
+    val customTypes = operation.fields.filter { it.fields.any() }.distinct().map {
+      // TODO: Also need to recurse into inner types
+      TypeSpec.interfaceBuilder(it.type)
```

Comment:
So the name of interface is based on the type of field, but there can be next case:

```
query CollectionsByHandle {
  shop {
    snowboards:collectionByHandle(handle: "snowboards") {
      id
      title
    }
    
    goggles:collectionByHandle(handle: "goggles") {
      id
      title
      bodyHtml
      updatedAt
    }
  }
}
```

that will produce next IR

```
...
"fields": [
						{
							"responseName": "snowboards",
							"fieldName": "collectionByHandle",
							"type": "Collection",
							"fields": [
								{
									"responseName": "id",
									"fieldName": "id",
									"type": "ID!"
								},
								{
									"responseName": "title",
									"fieldName": "title",
									"type": "String!"
								}
							],
							"fragmentSpreads": [],
							"inlineFragments": []
						},
						{
							"responseName": "goggles",
							"fieldName": "collectionByHandle",
							"type": "Collection",
							"fields": [
								{
									"responseName": "id",
									"fieldName": "id",
									"type": "ID!"
								},
								{
									"responseName": "title",
									"fieldName": "title",
									"type": "String!"
								},
								{
									"responseName": "bodyHtml",
									"fieldName": "bodyHtml",
									"type": "String"
								},
								{
									"responseName": "updatedAt",
									"fieldName": "updatedAt",
									"type": "DateTime!"
								}
							],
							"fragmentSpreads": [],
							"inlineFragments": []
						}
...
```

So both fields `snowboards` and `goggles` has the same type `Collection` but with different fields.
I guess `type` is not enough as the name of generated interface class, if there is alias it should be taken into account. So as example there should be generated 2 interfaces with names `SnowboardsCollection` and `GogglesCollection`